npm/src/main.ts

```diff
@@ -31,6 +31,15 @@ type Wasm = typeof import("../lib/node/qsc_wasm.cjs");
 let wasm: Wasm | null = null;
 const require = createRequire(import.meta.url);
 
+export async function provideTextDocumentContent(
```

Comment:
Minor nit - but you don't need the `async` keyword here. The function body doesn't `await` anything.

Reply:
(Though I guess that makes it easier to return the result as a Promise, so ignore this)